src/pf-touchspin/pf-touchspin.component.js

```diff
@@ -0,0 +1,387 @@
+      if (input !== document.activeElement) {
+        return;
+      }
+      event.stopPropagation();
+      event.preventDefault();
+      if (delta < 0) {
+        self._down();
+      } else {
+        self._up();
+      }
+    });
+
+    this._bindEvents();
+
+    this.initialized = true;
+    this.dispatchEvent(new CustomEvent('initialized', {}));
+  }
+
+  /**
+  * Called when element's attribute value has changed
+  *
+  * @param {string} attrName The attribute name that has changed
+  * @param {string} oldValue The old attribute value
+  * @param {string} newValue The new attribute value
+  */
+  attributeChangedCallback(attrName, oldValue, newValue) {
+    this.init();
+  }
+
+  /**
+   * Called when an instance of the element is created
+   */
+  constructor() {
+    super();
+  }
+
+  /**
+   *
+   */
+  _bindEvents() {
+    this.addEventListener('pf-touchspin.downonce', function () {
+      this._down();
+    });
+
+    this.addEventListener('pf-touchspin.uponce', function () {
+      this._up();
+    });
+
+    this.addEventListener('pf-touchspin.downspin', function () {
+      this._downSpin();
+    });
+
+    this.addEventListener('pf-touchspin.upspin', function () {
+      this._upSpin();
+    });
+
+    this.addEventListener('pf-touchspin.stop', function () {
+      this._stop();
+    });
+
+
+  }
+
+  /**
+   * force the valur to be divisible by step
+   *
+   * @param {number} value
+   */
+  _stepDivisibility(value) {
+    switch (this._forceStepDivisibility) {
+      case 'round':
+        return (Math.round(value / this._step) * this._step).toFixed(this._decimals);
+      case 'floor':
+        return (Math.floor(value / this._step) * this._step).toFixed(this._decimals);
+      case 'ceil':
+        return (Math.ceil(value / this._step) * this._step).toFixed(this._decimals);
+      default:
+        return value;
+    }
+  }
+
+  /**
+   * check the value before change in value
+   */
+  _checkValue() {
+    let val, parsedval, returnval;
+
+    val = this.querySelector('input').value;
+
+    if (this._decimals > 0 && val === '.') {
+      return;
+    }
+
+    parsedval = parseFloat(val);
+
+    if (isNaN(parsedval)) {
+      parsedval = 0;
+    }
+
+    returnval = parsedval;
+
+    if (parsedval < this.min) {
+      returnval = this.min;
+    }
+
+    if (parsedval > this.max) {
+      returnval = this.max;
+    }
+
+    returnval = this._stepDivisibility(returnval);
+
+    if (Number(val).toString() !== returnval.toString()) {
+      this.querySelector('input').value = returnval;
+    }
+  }
+
+
+  /**
+   * boost the value
+   *
+   * @param {number} value
+   */
+  _boostedStep(value) {
+    if (!this._booster) {
+      return this._step;
+    }
+    let boosted = Math.pow(2, Math.floor(this.spincount / this._boostat)) * this._step;
+
+    if (this._maxBoostedStep) {
+      if (boosted > this._maxBoostedstep) {
+        boosted = this._maxBoostedStep;
+        value = Math.round((value / boosted)) * boosted;
```

Comment:
Is this variable used ```value```? see it passed, then updated here, but unclear where it's used